Common/src/config_structure.cpp

```diff
@@ -153,9 +156,9 @@ unsigned short CConfig::GetnZone(string val_mesh_filename, unsigned short val_fo
 #ifndef HAVE_MPI
         exit(EXIT_FAILURE);
 #else
-        MPI_Barrier(MPI_COMM_WORLD);
-        MPI_Abort(MPI_COMM_WORLD,1);
-        MPI_Finalize();
+        SU2_MPI::Barrier(MPI_COMM_WORLD);
```

Comment:
Similar to my comment about rank/size, I would really like to see a cleaner, standard error handling. Doesn't have to be anything fancy, just a single routine, accessible everywhere (maybe in the MPI wrapper) that does something like the following:

SU2_MPI::Error("This is a string describing the error.", "CClassName::MethodName")

and inside will be the cout to dump the string and then the ifdef's for the correct exit in serial or parallel.

Reply:
Hi,

Following the same ideas, would it be possible to have a set/get mechanism for the communicator itself ? The CDriver takes a communicator as argument, but currently does nothing with it. The first motivation was to enable SU2 to be initialized by the Py wrapper with a specific communicator (e.g. a subset of MPI_COMM_WORLD). At the beginning the idea was to Set/Get this communicator from the CConfig class, but then we figured out that we were not able to access it everywhere in the code...if I remember well.

David

Reply:
I added a error function like you suggested @economon. Have a look at config_structure.cpp, I replaced already all of the "exit(...)" with that routine there.

@tobadavid, yep thats also something that should be possible.  I'll add a preliminary implementation soon.